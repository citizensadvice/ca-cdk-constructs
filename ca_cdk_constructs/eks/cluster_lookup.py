from aws_cdk import CfnOutput, Fn, Stack
from aws_cdk.aws_ec2 import IVpc
from aws_cdk.aws_eks import Cluster, ICluster, OpenIdConnectProvider
import aws_cdk.custom_resources as cr
from constructs import Construct


def get_eks_cluster(
    scope: Construct, env_name: str, kubectl_role_arn: str, vpc: IVpc
) -> ICluster:
    """_summary_

    Args:
        scope (Construct): scope
        env_name (str): environment name
        kubectl_role_arn (str): ARN of a role with deployment access to the namespace(s) used in this CDK project
        vpc (IVpc): The cluster VPC

    Returns:
        ICluster: _description_
    """
    cluster_name = {
        "test": "test-eks-controlplane",
        "qa": "devops-eks-controlplane",
        "staging": "prod-eks-controlplane",
        "prod": "prod-eks-controlplane",
    }[env_name]

    describe_cluster_sdk_call = cr.AwsSdkCall(
        service="EKS",
        action="describeCluster",
        physical_resource_id=cr.PhysicalResourceId.of(f"{env_name}ClusterOidcProviderLookup"),
        parameters={"name": cluster_name},
        output_paths=[
            "cluster.identity.oidc.issuer",
            "cluster.resourcesVpcConfig.clusterSecurityGroupId",
        ],
    )

    describe_cluster_cr = cr.AwsCustomResource(
        scope,
        "OidcConfigNameLookup",
        on_create=describe_cluster_sdk_call,
        on_update=describe_cluster_sdk_call,
        policy=cr.AwsCustomResourcePolicy.from_sdk_calls(
            resources=cr.AwsCustomResourcePolicy.ANY_RESOURCE
        ),
    )

    oidc_provider_arn = Fn.join(
        "",
        [
            f"arn:aws:iam::{Stack.of(scope).account}:oidc-provider/",
            Fn.select(
                1,
                Fn.split(
                    "https://",
                    describe_cluster_cr.get_response_field("cluster.identity.oidc.issuer"),
                ),
            ),
        ],
    )

    sg_id = describe_cluster_cr.get_response_field(
        "cluster.resourcesVpcConfig.clusterSecurityGroupId"
    )

    private_subnets = list(map(lambda s: s.subnet_id, vpc.private_subnets))
    oidc_provider = OpenIdConnectProvider.from_open_id_connect_provider_arn(
        scope, "ClusterOIDCProvider", open_id_connect_provider_arn=oidc_provider_arn
    )

    # useful to debug sdk lookup calls
    CfnOutput(scope, "EksClusterSecurityGroupId", value=sg_id)
    CfnOutput(scope, "EksClusterOidcProviderArn", value=oidc_provider_arn)

    return Cluster.from_cluster_attributes(
        scope,
        cluster_name,
        kubectl_role_arn=kubectl_role_arn,
        open_id_connect_provider=oidc_provider,
        cluster_name=cluster_name,
        kubectl_private_subnet_ids=private_subnets,
        kubectl_security_group_id=sg_id,
        vpc=vpc,
    )
