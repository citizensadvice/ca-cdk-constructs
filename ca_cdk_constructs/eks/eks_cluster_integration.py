from typing import Optional
import aws_cdk.custom_resources as cr
from aws_cdk import CfnOutput, Fn, Stack
from aws_cdk.aws_ec2 import IVpc
from aws_cdk.aws_eks import Cluster, OpenIdConnectProvider
from aws_cdk.aws_iam import IRole, Role, AccountRootPrincipal
from aws_cdk.lambda_layer_kubectl_v28 import KubectlV28Layer as KubectlLayer
from constructs import Construct


class EksClusterIntegration(Construct):
    """Looks up an existing EKS cluster by name and provisions (or uses an existing) kubectl IAM role needed for deployments to the cluster."""

    def __init__(
        self,
        scope: Construct,
        id: str,
        vpc: IVpc,
        cluster_name: str,
        role_name: Optional[str] = None,
        role: Optional[IRole] = None,
        prune: bool = True,
    ):
        """

        Args:
            scope (Construct): parent Construct
            id (str): id
            vpc (IVpc): EC2 Vpc, needed to lookup the cluster
            cluster_name (str): the name of the EKS cluster
            role_name (str, optional): The name of the kubectl IAM role that will be created to enable deployments to the cluster. Defaults to None.
            role (IRole, optional): Existing kubectl IAM role to use for deployments to the cluster. Either this or role_name must be set. Defaults to None.
            prune (bool, optional): Indicates whether Kubernetes resources added through ``addManifest()`` can be automatically pruned. Defaults to True.

        Raises:
            Exception: if neither or both of role_name, role are set
        """
        super().__init__(scope, id)

        if not bool(role) ^ bool(role_name):
            raise Exception("Either role or role_name must be set")

        self.role = role or Role(
            self,
            "KubernetesAuthRole",
            assumed_by=AccountRootPrincipal(),
            role_name=role_name,
            description="Role used by the AWS CDK project to authenticate with Kubernetes",
        )

        describe_cluster_sdk_call = cr.AwsSdkCall(
            service="EKS",
            action="describeCluster",
            physical_resource_id=cr.PhysicalResourceId.of("DescribeClusterLookup"),
            parameters={"name": cluster_name},
            output_paths=[
                "cluster.identity.oidc.issuer",
                "cluster.resourcesVpcConfig.clusterSecurityGroupId",
            ],
        )

        # we need the OIDC provider in order to deploy Service accounts
        # to the imported cluster
        describe_cluster_cr = cr.AwsCustomResource(
            self,
            "OidcConfigNameLookup",
            on_create=describe_cluster_sdk_call,
            on_update=describe_cluster_sdk_call,
            policy=cr.AwsCustomResourcePolicy.from_sdk_calls(
                resources=cr.AwsCustomResourcePolicy.ANY_RESOURCE
            ),
        )

        oidc_provider_arn = Fn.join(
            "",
            [
                f"arn:aws:iam::{Stack.of(self).account}:oidc-provider/",
                Fn.select(
                    1,
                    Fn.split(
                        "https://",
                        describe_cluster_cr.get_response_field("cluster.identity.oidc.issuer"),
                    ),
                ),
            ],
        )

        sg_id = describe_cluster_cr.get_response_field(
            "cluster.resourcesVpcConfig.clusterSecurityGroupId"
        )

        private_subnets = [s.subnet_id for s in vpc.private_subnets]
        oidc_provider = OpenIdConnectProvider.from_open_id_connect_provider_arn(
            self, "ClusterOIDCProvider", open_id_connect_provider_arn=oidc_provider_arn
        )

        self.cluster = Cluster.from_cluster_attributes(
            self,
            "Resource",
            kubectl_role_arn=self.role.role_arn,
            open_id_connect_provider=oidc_provider,
            cluster_name=cluster_name,
            kubectl_private_subnet_ids=private_subnets,
            kubectl_security_group_id=sg_id,
            kubectl_layer=KubectlLayer(self, "KubectlLayer"),
            vpc=vpc,
            prune=prune,  # https://github.com/aws/aws-cdk/issues/19843
        )

        # useful to debug sdk lookup calls
        CfnOutput(self, "EksClusterSecurityGroupId", value=sg_id)
        CfnOutput(self, "EksClusterOidcProviderArn", value=oidc_provider_arn)

        CfnOutput(self, "K8sAuthRoleName", description="Role name", value=self.role.role_name)
        CfnOutput(self, "K8sAuthRoleArn", description="Role ARN", value=self.role.role_arn)
