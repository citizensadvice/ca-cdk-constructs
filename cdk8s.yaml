language: python
imports:
  - k8s@1.26.0
  # must match the version deployed to the cluster
  - https://raw.githubusercontent.com/external-secrets/external-secrets/v0.6.1/deploy/crds/bundle.yaml
