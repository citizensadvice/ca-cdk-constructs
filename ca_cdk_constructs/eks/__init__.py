from ca_cdk_constructs.eks.eks_cluster_integration import EksClusterIntegration
