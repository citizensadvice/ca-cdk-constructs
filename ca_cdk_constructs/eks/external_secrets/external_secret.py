from enum import Enum
from typing import Any

from aws_cdk.aws_secretsmanager import ISecret
from aws_cdk.aws_ssm import StringParameter
from constructs import Construct

from ca_cdk_constructs.eks.imports.io.external_secrets import (
    ExternalSecretV1Beta1,
    ExternalSecretV1Beta1Spec,
    ExternalSecretV1Beta1SpecSecretStoreRef,
    ExternalSecretV1Beta1SpecTarget,
    ExternalSecretV1Beta1SpecData,
    ExternalSecretV1Beta1SpecDataRemoteRef,
)


class ExternalSecretStore(Enum):
    """Known, existing ExternalSecret Stores, created by cluster admins and having predefined names"""

    VAULT = "vault"
    AWS_SSM = "aws-secrets-manager"
    AWS_PARAMETER_STORE = "aws-parameter-store"


class IExternalSecretSource:
    """Convenience class to collect external secrets configurations"""

    def __init__(
        self, k8s_secret_name: str, source_secret: Any, secret_mappings: dict[str, str]
    ):
        """

        Args:
            k8s_secret_name (str): The name of the Kubernetes secret generated by ExternalSecrets
            source_secret (Any): the external secret source, e.g. Vault path or AWS SM secret object
            secret_mappings (dict[str, str]): Dictionary structure describing how to map the data in the external secret to env variables. The structure is:
                ``{ <secret key>: <env variable name> }``
            e.g.
                { "password": "DB_PASSWORD" }
            To use the secret key as env variable name, set the value to None or empty string e.g.
                ``{ "password": "" } # will set env var 'password'```
        """
        self.k8s_secret_name = k8s_secret_name
        self.source_secret = source_secret
        self.secret_mappings = secret_mappings

    def secret_store(self) -> ExternalSecretStore:
        return None

    def secret_source_id(self) -> str:
        """
        The secret source ID which will be set in spec.data.remoteRef.key
        @see https://external-secrets.io/v0.5.7/spec/#external-secrets.io/v1beta1.ExternalSecretDataRemoteRef
        """
        return str(self.source_secret)


class ExternalVaultSecret(IExternalSecretSource):
    def secret_store(self):
        return ExternalSecretStore.VAULT


class ExternalAWSSMSecret(IExternalSecretSource):
    def __init__(self, k8s_secret_name: str, source_secret: ISecret, secret_mappings: dict[str, str]):
        super().__init__(k8s_secret_name, source_secret, secret_mappings)

    def secret_source_id(self) -> str:
        return self.source_secret.secret_name

    def secret_store(self) -> ExternalSecretStore:
        return ExternalSecretStore.AWS_SSM


class ExternalAWSParameterStoreSecret(IExternalSecretSource):
    def __init__(
        self, k8s_secret_name: str, source_secret: StringParameter, secret_mappings: dict[str, str]
    ):
        super().__init__(k8s_secret_name, source_secret, secret_mappings)

    def secret_source_id(self) -> str:
        return self.source_secret.parameter_name

    def secret_store(self) -> ExternalSecretStore:
        return ExternalSecretStore.AWS_SSM


class ExternalSecret(Construct):
    """
    io.external_secrets.ExternalSecret facade which:
        - provides a simpler interface for creating known, supported ExternalSecret configurations, e.g. it does not support custom [templating](https://external-secrets.io/v0.7.0-rc1/guides/templating/)
        - always uses an ExternalSecrets API version that is compatible and supported in CA - managed clusters.
        - is restricted to only some supported SecretStores - `~ca_cdk_constructs.eks.external_secrets.ExternalSecretStore`

    Usage:

        ExternalSecret(self, "secret",
            k8s_secret_name="app-secret",
            secret_store=ExternalSecretStore.VAULT,
            source_secret="path/in/vault",
            secret_mappings={
                "key_in_vault": "KEY_IN_K8S_SECRET",
                "key_in_vault.subkey": "KEY_IN_K8S_SECRET2",
                "key_in_vault2": "", # sets key in secret = key in external secret, i.e 'key_in_vault2'
            }
        )

        ExternalSecret(self, "secret2",
            k8s_secret_name="app-secret2",
            secret_store=ExternalSecretStore.AWS_PARAMETER_STORE,
            source_secret="secret-1", # AWS parameter store parameter name
            secret_mappings={
                "some_key": "KEY_IN_K8S_SECRET",
                "some_key.subkey": "FOO",
            }
        )
    """

    def __init__(
        self,
        scope: Construct,
        id: str,
        k8s_secret_name: str,
        secret_store: ExternalSecretStore,
        source_secret: str,
        secret_mappings: dict[str, str],
        metadata={},
    ):
        super().__init__(scope, id)
        self._k8s_secret_name = k8s_secret_name
        self._source_secret = source_secret
        self._secret_store = secret_store

        ExternalSecretV1Beta1(
            self,
            "Resource",
            metadata=metadata,
            spec=ExternalSecretV1Beta1Spec(
                secret_store_ref=ExternalSecretV1Beta1SpecSecretStoreRef(
                    name=secret_store.value, kind="SecretStore"  # known, existing store
                ),
                target=ExternalSecretV1Beta1SpecTarget(name=k8s_secret_name),
                data=[
                    ExternalSecretV1Beta1SpecData(
                        remote_ref=ExternalSecretV1Beta1SpecDataRemoteRef(
                            # the source ID, e.g. AWS SSM secret name or Vault path
                            key=source_secret,
                            # which property to retrieve from provider
                            property=k,
                        ),
                        secret_key=v  # the key in the K8s secret e.g. the env var name
                        or str(k).split(".")[-1],
                        # if not specified, set the secret key = the last element of the pottentially . separated provider key
                    )
                    for k, v in secret_mappings.items()
                ],
            ),
        )

    @property
    def k8s_secret_name(self) -> str:
        return self._k8s_secret_name

    @property
    def source_secret(self) -> str:
        return self._source_secret

    @property
    def secret_store(self) -> ExternalSecretStore:
        return self._secret_store
