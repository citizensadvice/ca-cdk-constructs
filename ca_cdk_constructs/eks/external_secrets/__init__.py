from .external_secrets_aws_secret_store import ExternalSecretsAwsSecretStore  # noqa: F401
from .external_secret import ExternalSecretSource, ExternalSecret  # noqa: F401
from .external_aws_secrets_chart import ExternalAwsSecretsChart  # noqa: F401
