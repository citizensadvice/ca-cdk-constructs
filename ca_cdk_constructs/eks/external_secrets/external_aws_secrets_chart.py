from cdk8s import Chart
from constructs import Construct

from ca_cdk_constructs.eks.external_secrets import ExternalSecretSource
from ca_cdk_constructs.eks.external_secrets.external_secret import ExternalSecret
from ca_cdk_constructs.eks.external_secrets.external_secrets_aws_secret_store import (
    ExternalSecretsAwsSecretStore,
)
from ca_cdk_constructs.eks.imports.io.external_secrets import (
    SecretStoreSpecProviderAwsService,
)


class ExternalAwsSecretsChart(Chart):
    """
    cdk8s_lib chart for deploying ExternalSecrets from AWS SecretsManager or ParameterStore.
    The chart deploys a SecretStore object of the specified type (AWS SSM or ParameterStore)
    and one or more external secrets linked to the store.
    The chart assumes that access to the secrets is managed by an existing `ServiceAccount`.

    The secret types ( set in `~secret_sources~` ) must match the store type e.g setting `~secret_service~` to `SecretStoreSpecProviderAwsService.SECRETS_MANAGER`
    requires a list of `~ExternalSecretSource~`s where each `ExternalSecretSource` references an AWS SSM secret.

    To use secrets from both AWS SSM and Parameter store,
    deploy two instances of the chart, setting `~secret_service~` to the relevant value
    and passing a list of compatible secret sources.
    """

    def __init__(
        self,
        scope: Construct,
        id: str,
        *,
        service_account_name: str,
        region: str,
        secret_sources: list[ExternalSecretSource],
        secret_service: SecretStoreSpecProviderAwsService = SecretStoreSpecProviderAwsService.SECRETS_MANAGER,
        disable_resource_name_hashes: bool = False,
        labels: dict[str, str] = {},
        namespace: str = "",
    ):
        """
        Args:
            scope (Construct): _description_
            id (str): _description_
            service_account_name (str): the name of the `ServiceAccount` used by the `SecretStore` to access AWS secret services
            region (str): AWS region
            secret_sources (list[ExternalSecretSource]):  The secret sources to generate k8s secrets for
            secret_service (SecretStoreSpecProviderAwsService, optional): The service type which defines where to fetch the secrets_from. Defaults to SecretStoreSpecProviderAwsService.SECRETS_MANAGER.
            disable_resource_name_hashes: The autogenerated resource name by default is suffixed with a stable hash of the construct path. Setting this property to true drops the hash suffix. Default: false
            labels: Labels to apply to all resources in this chart. Default: - no common labels
            namespace: The default namespace for all objects defined in this chart (directly or indirectly). This namespace will only apply to objects that don't have a ``namespace`` explicitly defined for them. Default: - no namespace is synthesized (usually this implies "default")
        """
        labels.setdefault("app.kubernetes.io/managed-by", "aws-cdk")
        labels.setdefault("app.kubernetes.io/component", "external-secrets")

        super().__init__(
            scope,
            id,
            disable_resource_name_hashes=disable_resource_name_hashes,
            labels=labels,
            namespace=namespace,
        )

        self._k8s_secret_names = [s.k8s_secret_name for s in secret_sources]

        store = ExternalSecretsAwsSecretStore(
            self,
            "SecretStore",
            region=region,
            service_account_name=service_account_name,
            secret_service=secret_service,
        )

        for secret_spec in secret_sources:
            ExternalSecret(
                self,
                f"{secret_spec.k8s_secret_name}-external-secret",
                secret_source=secret_spec,
                store_name=store.name,
            )

    @property
    def k8s_secret_names(self) -> list[str]:
        return self._k8s_secret_names
