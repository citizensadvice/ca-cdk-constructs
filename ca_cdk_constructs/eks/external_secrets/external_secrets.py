from typing import List
from aws_cdk import Stack

from cdk8s import Chart
from aws_cdk.aws_eks import KubernetesPatch, ICluster
import cdk8s_plus_21
from constructs import Construct

from ca_cdk_constructs.eks.external_secrets.external_secret import ExternalSecret
from ca_cdk_constructs.eks.external_secrets.external_secret_source import (
    ExternalSecretSource,
)


class ExternalSecrets(Construct):
    """Collect external secret from AWS Secrets Manager or Vault.

    Requires existing external secrets SecretStore(s), possibly created by a different workflow as explained in https://external-secrets.io/v0.5.8/api-overview/#secretstore

    ## Example

    ```
    from ca_cdk_constructs.eks.external_secrets import ExternalSecrets, ExternalVaultSecretSource, ExternalAWSSMSecretSource

    app = k8s.App()
    chart = Chart(app, "Myservice")

    # a SSM secret that needs passing to the pods
    secr = Secret(...)
    role = Role(...)
    secr.grant_read(role)
    # IRSA to access the secret
    ServiceAccount(
            self,
            "ExternalSecretsKubernetesServiceAccount",
            namespace="myapp-namespace",
            name="external-secrets-aws", # match the name in the SecretStore, see above
            cluster=cluster
    )


    # define the external secrets
    secrets=ExternalSecrets(self, "ExternalSecrets",
        chart=chart,
        secret_sources=[
            ExternalVaultSecret(
                k8s_secret_name="vault",
                secret="devops/newrelic",
                secret_mappings={"key": "NEW_RELIC_KEY"},
            ),
            ExternalAWSSMSecret(
                k8s_secret_name="secretsmanager-secret",
                secret=secr,
                secret_mappings={"key": "NEW_RELIC_KEY", "BLAH": ""},
            ),
        ],
    )
    ```

    ###  Using with cdk8s

    ```
    deployment = Deployment(chart, ........)

    ## add to cdk8s containers
    secrets.add_to_containers(containers=deployment.containers)
    ```

    ## Using with helm, requires that the chart exposes overrides for the generated secrets
    ```
    helm_overrides = {
        "secrets" : secrets.k8s_secret_names
    }
    ```

    ## finally, deploy the chart
    ```
    cluster.add_cdk8s_chart("ExternalSecretsDeployment", secrets.chart)

    ```
    """

    def __init__(
        self,
        scope: Construct,
        id: str,
        chart: Chart,
        secret_sources: List[ExternalSecretSource],
    ) -> None:
        super().__init__(scope, id)
        self.chart = chart
        self.secret_sources = secret_sources
        for secret_source in self.secret_sources:
            self.create_external_secret(secret_source)

    def add_to_containers(self, containers: list[cdk8s_plus_21.Container]):
        for container in containers:
            for secret_source in self.secret_sources:
                self.add_secret_to_container(container, secret_source)

    def create_external_secret(self, secret_source: ExternalSecretSource) -> ExternalSecret:
        return ExternalSecret(
            self.chart,
            f"{secret_source.k8s_secret_name}ExternalSecret",
            k8s_secret_name=secret_source.k8s_secret_name,
            secret_store=secret_source.secret_store(),
            secret_mappings={secret_source.secret_source_id(): secret_source.secret_mappings},
        )

    def add_secret_to_container(
        self, container: cdk8s_plus_21.Container, secret_source: ExternalSecretSource
    ):
        container.env.copy_from(
            cdk8s_plus_21.Env.from_secret(
                cdk8s_plus_21.Secret.from_secret_name(name=secret_source.k8s_secret_name)
            )
        )

    @property
    def k8s_secret_names(self) -> list[str]:
        """
        Returns:
            list[str]: a list containing the names of the k8s secrets that would be created by ExternalSecrets
        """
        return [s.k8s_secret_name for s in self.secret_sources]

    @property
    def container_secret_refs(self) -> list[dict[str, str]]:
        """
        Returns:
            list[dict[str,str]]: a list of container `envFrom` references to all k8s secrets created by ExternalSecrets
        """
        return [{"secretRef": s.k8s_secret_name} for s in self.secret_sources]
