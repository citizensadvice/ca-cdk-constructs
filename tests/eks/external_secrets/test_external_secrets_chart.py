from cdk8s import App

from ca_cdk_constructs.eks.external_secrets import (
    ExternalSecretSource,
)
from ca_cdk_constructs.eks.external_secrets.external_aws_secrets_chart import (
    ExternalAwsSecretsChart,
)
from ca_cdk_constructs.eks.imports.io.external_secrets import (
    SecretStoreV1Beta1SpecProviderAwsService as SecretStoreSpecProviderAwsService,
)


# to update the snapshot, run:
#   poetry run pytest tests/cdk8s_lib/external_secrets/test_external_secrets_chart.py --snapshot-update
def test_external_secret_chart(snapshot):
    app = App()

    source = ExternalSecretSource(
        source_secret="db-secret",
        secret_mappings={"username": "DB_USER"},
        k8s_secret_name="db-secret",
    )

    ExternalAwsSecretsChart(
        app,
        "ExternalSecretsDeployment",
        region="eu-west-1",
        service_account_name="SA",  # not in scope
        secret_service=SecretStoreSpecProviderAwsService.SECRETS_MANAGER,
        secret_sources=[source],
        namespace="test-ns",
    )
    assert app.synth_yaml() == snapshot
