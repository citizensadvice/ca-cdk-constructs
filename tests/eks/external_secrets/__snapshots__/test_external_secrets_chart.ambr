# serializer version: 1
# name: test_external_secret_chart
  '''
  apiVersion: external-secrets.io/v1beta1
  kind: SecretStore
  metadata:
    labels:
      app.kubernetes.io/component: external-secrets
      app.kubernetes.io/managed-by: aws-cdk
    name: externalsecretsdeployment-secretstore-c8c6b318
    namespace: test-ns
  spec:
    provider:
      aws:
        auth:
          jwt:
            serviceAccountRef:
              name: SA
        region: eu-west-1
        service: SecretsManager
  ---
  apiVersion: external-secrets.io/v1beta1
  kind: ExternalSecret
  metadata:
    labels:
      app.kubernetes.io/component: external-secrets
      app.kubernetes.io/managed-by: aws-cdk
    name: externalsecretsdepl-db-secret-external-secret-c8e9965e
    namespace: test-ns
  spec:
    data:
      - remoteRef:
          key: db-secret
          property: username
        secretKey: DB_USER
    refreshInterval: 1h
    secretStoreRef:
      kind: SecretStore
      name: externalsecretsdeployment-secretstore-c8c6b318
    target:
      name: db-secret
  
  '''
# ---
